src/utils/featheryClient/index.ts

```diff
@@ -474,6 +530,53 @@ export default class FeatheryClient extends IntegrationClient {
     );
   }
 
+  /**
+   * `beforeunload` event handler that flushes the pending submit custome changes
+   * when a user is attempting to exit the page.
+   * @param event `BeforeUnloadEvent`
+   * @returns
+   */
+  _flushPendingChangesBeforeUnload(event: BeforeUnloadEvent) {
+    event.preventDefault();
+    this.flushPendingSubmitCustomUpdates();
+    return (event.returnValue = '');
+  }
+
+  async submitCustom(
+    customKeyValues: { [key: string]: any },
+    options?: {
+      override?: boolean;
```

Comment:
customKeyValues, {override=true, shouldFlush=false}

destructure directly with defaults will make this cleaner